### PR TITLE
Remove methods from PayloadConvention that are only needed by the DefaultPayloadConvention class

### PR DESCRIPTION
**Pull request by drwill-ms**
We can simplify the PayloadConvention base class by removing 2 methods that are really only required on the DefaultPayloadConvention class.

### CODE REVIEW

File: iothub/device/src/Pipeline/AmqpTransportHandler.cs [hunk 1]
```diff
@@ -372,7 +372,7 @@ public override async Task<TwinProperties> GetTwinAsync(CancellationToken cancel
                 {
                     // The response here is deserialized into an SDK-defined type based on service-defined NewtonSoft.Json-based json property name.
                     // For this reason, we use NewtonSoft Json serializer for this deserialization.
-                    TwinDocument clientTwinProperties = DefaultPayloadConvention.Instance.GetObject<TwinDocument>(responseFromService.BodyStream);
+                    TwinDocument clientTwinProperties = DefaultPayloadConvention.GetObject<TwinDocument>(responseFromService.BodyStream);
```
Comment (drwill-ms) [inline iothub/device/src/Pipeline/AmqpTransportHandler.cs:375]: These can become static methods

File: iothub/device/src/Serialization/PayloadConvention.cs [hunk 2]
```diff
@@ -2,14 +2,16 @@
...
 
 using System.IO;
-using System.Text;
-using System.Threading.Tasks;
 
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// The payload convention class.
+    /// The payload convention class intended for serializing and deserializing JSON payloads.
```
Comment (drwill-ms) [inline iothub/device/src/Serialization/PayloadConvention.cs:9]: Specify we intend this for JSON payloads.

File: iothub/device/src/Serialization/SystemTestJsonPayloadConvention.cs [hunk 3]
```diff
@@ -41,19 +40,5 @@ public override T GetObject<T>(byte[] objectToConvert)
         {
             return JsonSerializer.Deserialize<T>(objectToConvert);
         }
-
-        /// <inheritdoc/>
-        public override T GetObject<T>(Stream streamToConvert)
```
Comment (drwill-ms) [inline iothub/device/src/Serialization/SystemTestJsonPayloadConvention.cs:46]: No longer needed here.

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**